### PR TITLE
Friends JSON API Resource

### PR DESCRIPTION
**Pull request by ultimatemonty**
WIP

### CODE REVIEW

File: app/resources/jsonapi/user_resource.rb [hunk 1]
```diff
@@ -1,7 +1,9 @@
 class Jsonapi::UserResource < JSONAPI::Resource
   attributes :avatar_url, :email, :first_name, :last_name,
      :middle_name, :password
+
   has_many :scorecards
+  has_many :friends, class_name: 'Friend'
```
Comment (gschorkopf) [inline app/resources/jsonapi/user_resource.rb]: The class name shouldn't be necessary... interesting if it is.

**Comment by gschorkopf** [comment]
Closing for now. Let's discuss what to do with the concept of "friends".

I'd kinda prefer being able to search for anyone and not have to add a friend first. There might be value in being able to only view certain people's scorecards on the "my friends' rounds" section, but let's worry about that later.

**Comment by ultimatemonty** [comment]
:+1: my basic want for "friends" is to be able to identify players that i play with on a regular basis and then surface those as "recommended" options with the ability to still search for and select any other user. Maybe group by:
* Recent
* Frequent
* All

**Comment by gschorkopf** [comment]
Yeah, I like the idea of recent and frequent too. We could use the existing friends' model to make that work. Maybe `tap` (meaning adjust the `updated_at` field) whenever a new round with that friend is started and order by that field for recent. Maybe have a field on the join table called `rounds_played_together` or something for frequent. We can calculate these on new round creation or daily, either would probably suffice.

**Pull request closed without merging**